builtin/providers/aws/resource_aws_ami_launch_permission_test.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	r "github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"testing"
+)
+
+func TestAccAWSAMILaunchPermission_Basic(t *testing.T) {
+	image_id := ""
+	account_id := os.Getenv("AWS_ACCOUNT_ID")
+
+	r.Test(t, r.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if os.Getenv("AWS_ACCOUNT_ID") == "" {
+				t.Fatal("AWS_ACCOUNT_ID must be set")
+			}
+		},
+		Providers: testAccProviders,
+		Steps: []r.TestStep{
+			// Scaffold everything
+			r.TestStep{
+				Config: testAccAWSAMILaunchPermissionConfig(account_id, true),
+				Check: r.ComposeTestCheckFunc(
+					testCheckResourceGetAttr("aws_ami_copy.test", "id", &image_id),
+					testAccAWSAMILaunchPermissionExists(account_id, &image_id),
+				),
+			},
+			// Drop just launch permission to test destruction
+			r.TestStep{
+				Config: testAccAWSAMILaunchPermissionConfig(account_id, false),
+				Check: r.ComposeTestCheckFunc(
+					testAccAWSAMILaunchPermissionDestroyed(account_id, &image_id),
+				),
+			},
+		},
+	})
+}
+
+func testCheckResourceGetAttr(name, key string, value *string) r.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		is := rs.Primary
+		if is == nil {
+			return fmt.Errorf("No primary instance: %s", name)
+		}
+
+		*value = is.Attributes[key]
+		return nil
+	}
+}
+
+func testAccAWSAMILaunchPermissionExists(account_id string, image_id *string) r.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+		if has, err := hasLaunchPermission(conn, *image_id, account_id); err != nil {
+			return err
+		} else if !has {
+			return fmt.Errorf("launch permission does not exist for '%s' on '%s'", account_id, *image_id)
+		}
+		return nil
+	}
+}
+
+func testAccAWSAMILaunchPermissionDestroyed(account_id string, image_id *string) r.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+		if has, err := hasLaunchPermission(conn, *image_id, account_id); err != nil {
+			return err
+		} else if has {
+			return fmt.Errorf("launch permission still exists for '%s' on '%s'", account_id, *image_id)
+		}
+		return nil
+	}
+}
+
+func testAccAWSAMILaunchPermissionConfig(account_id string, includeLaunchPermission bool) string {
+	base := `
+provider "aws" {
```

Comment:
Thanks! Dropping. I originally was using an AMI in us-east-1, but upon massive changes yesterday, forgot to remove that block.